packages/compose/src/hooks/use-media-query/test/index.js

```diff
@@ -1,136 +1,80 @@
 /**
  * External dependencies
  */
-import { render, act } from '@testing-library/react';
+import { act, render } from '@testing-library/react';
+import {
+	matchMedia,
+	MediaQueryListEvent,
+	setMedia,
+	cleanup,
+} from 'mock-match-media';
 
 /**
  * Internal dependencies
  */
 import useMediaQuery from '../';
 
-describe( 'useMediaQuery', () => {
-	let addListener, removeListener;
+const TestComponent = ( { query } ) => {
+	const queryResult = useMediaQuery( query );
+	return `useMediaQuery: ${ queryResult }`;
+};
 
+describe( 'useMediaQuery', () => {
 	beforeAll( () => {
-		jest.spyOn( global, 'matchMedia' );
-
-		addListener = jest.fn();
-		removeListener = jest.fn();
+		window.matchMedia = matchMedia;
+		window.MediaQueryListEvent = MediaQueryListEvent;
```

Comment:
Do you need to set the `MediaQueryListEvent` global? That seems useful only if you wanted to manually create and dispatch these events.